org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/findReferences/WorkspaceResourceAccess.xtend

```diff
@@ -25,7 +25,11 @@ class WorkspaceResourceAccess implements IResourceAccess {
 
 	override <R> readOnly(URI targetURI, IUnitOfWork<R, ResourceSet> work) {
 		return workspaceManager.doRead(targetURI) [ document, resource |
-			return work.exec(resource.resourceSet)
+			val resourceSet = resource?.resourceSet
+			if (resourceSet === null) {
+				return null
```

Comment:
We return `null` in all other implementation if `resourceSet` cannot be found, that's why I've put a comment about it the javadoc for `ResourceAccess` and used such guard, e.g. https://github.com/eclipse/xtext-eclipse/blob/242b5fcc83f4697f6381f9264f710773e8435ce8/org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/findrefs/LoadingResourceAccess.java#L46. 

But I think you are right to return `null` only when `resource === null` here and fail in other cases.